runtime/queries/perl/highlights.scm

```diff
@@ -1,181 +1,87 @@
+(heredoc_content) @string
+(heredoc_end) @string.special
+[(escape_sequence) (escaped_delimiter)] @string.special
 
-[
-"("
-")"
-"["
-"]"
-"{"
-"}"
-] @punctuation.bracket
-(standard_input_to_variable) @punctuation.bracket
+[(quoted_regexp) (match_regexp)] @string.regexp
 
-[
-"=~"
-"or"
-"="
-"=="
-"+"
-"-"
-"."
-"//"
-"||"
-(arrow_operator)
-(hash_arrow_operator)
-(array_dereference)
-(hash_dereference)
-(to_reference)
-(type_glob)
-(hash_access_variable)
-(ternary_expression)
-(ternary_expression_in_hash)
-] @operator
+(autoquoted_bareword _?) @string.special
 
-[
-(regex_option)
-(regex_option_for_substitution)
-(regex_option_for_transliteration)
-] @variable.parameter
-
-(type_glob
-  (identifier) @variable)
-(
-  (scalar_variable)
-  .
-  ("->" @operator))
+[(scalar) (arraylen)] @variable.scalar
+(scalar_deref_expression ["->" "$" "*"] @variable.scalar)
+(array) @variable.array
+(array_deref_expression ["->" "@" "*"] @variable.array)
+(hash) @variable.hash
+(hash_deref_expression ["->" "%" "*"] @variable.hash)
 
-[
-(word_list_qw)
-(command_qx_quoted)
-(string_single_quoted)
-(string_double_quoted)
-(string_qq_quoted)
-(bareword)
-(transliteration_tr_or_y)
-] @string
+(array_element_expression [array:(_) "->" "[" "]"] @variable.array)
+(slice_expression [array:(_) "->" "[" "]"] @variable.array)
+(keyval_expression [array:(_) "->" "[" "]"] @variable.array)
 
-[
-(regex_pattern_qr) 
-(patter_matcher_m)
-(substitution_pattern_s)
-] @string.regexp
+(hash_element_expression [hash:(_) "->" "{" "}"] @variable.hash)
+(slice_expression [hash:(_) "->" "[" "]"] @variable.hash)
+(keyval_expression [hash:(_) "->" "[" "]"] @variable.hash)
 
-(escape_sequence) @string.special
+(hash_element_expression key: (bareword) @string.special)
 
-[
-","
-(semi_colon)
-(start_delimiter)
-(end_delimiter)
-(ellipsis_statement)
-] @punctuation.delimiter
+(use_statement (package) @type)
+(package_statement (package) @type)
+(require_expression (bareword) @type)
 
-[
-(integer)
-(floating_point)
-(scientific_notation)
-(hexadecimal)
-] @constant.numeric
+(subroutine_declaration_statement name: (_) @function)
+(attrlist (attribute) @decorator)
 
-[
-; (if_statement)
-(unless_statement)
-(if_simple_statement)
-(unless_simple_statement)
-] @keyword.control.conditional
+(goto_expression (label) @label)
+(loopex_expression (label) @label)
 
-[
-"if"
-"elsif"
-"else"
-] @keyword.control.conditional 
+(statement_label label: _ @label)
+
+(relational_expression operator: "isa" right: (bareword) @type)
 
-(foreach_statement) @keyword.control.repeat
-(foreach_statement
-  .
-  (scope) @keyword)
+(function_call_expression (function) @function)
+(method_call_expression (method) @function.method)
+(method_call_expression invocant: (bareword) @type)
 
-(function_attribute) @label
+(func0op_call_expression function: _ @function.builtin)
+(func1op_call_expression function: _ @function.builtin)
 
-(function_signature) @type
+(function) @function
 
+(ERROR) @error
```

Comment:
```suggestion
```

In the future we'd like to add an option for error highlighting but by default we leave it off because it can be visually distracting